plutus-core/plutus-ir/test/transform/inline/single.golden

```diff
@@ -9,7 +9,15 @@
   (termbind
     (strict)
     (vardecl trivialLambda (con integer))
-    [ (lam y (con integer) y) [ (lam y (con integer) y) (con integer 1) ] ]
+    (let
+      (nonrec)
+      (termbind
+        (strict)
+        (vardecl f (fun (con integer) (con integer)))
+        (lam y (con integer) y)
+      )
+      [ f [ f (con integer 1) ] ]
+    )
```

Comment:
Not really. This PR stops allowing `LamAbs` and `TyAbs` to be inlined (except for `trace` and `error`). This could result in increased program sizes in some cases, but could also do the opposite in some other cases.